torch/_inductor/decomposition.py

```diff
@@ -741,3 +741,76 @@ def max_pool2d_with_indices(
         padding,
     )
     return vals, indices
+
+
+def _use_online_softmax(x, dim):
+    if not config.online_softmax:
+        return False
+
+    # Don't do online softmax for scalar or 1d tensor
+    if x.dim() < 2:
+        return False
+
+    # Only do online softmax for GPU for now
+    if x.device.type != "cuda":
+        return False
+
```

Comment:
add comment why this particular number?  could we also make this a config option so that we can test it more easily on lower sizes ?